src/test/csf/BasicNetwork.h

```diff
@@ -136,9 +143,19 @@ class BasicNetwork
     */
     bool
     connect(
-        Peer const& from,
-        Peer const& to,
-        duration const& delay = std::chrono::seconds{0});
+            Peer const& from,
+            Peer const& to,
+            duration const& delay = std::chrono::seconds{0})
+    {
+        connect(from, to,
+                DurationDistributionRef(ConstantDuration{delay}));
+    }
+
+    bool
+    connect(
+            Peer const& from,
+            Peer const& to,
+            DurationDistributionRef delayGen);
```

Comment:
Since the link constructor is templated on `DurationDistribution`, should this be too?  I like that idea for all the various layers of connect; template all the way down except the last layer in connect, which stores things into the link as a type-erased holder.